### ISSUE TITLE
Jest testing error. TypeError: Cannot assign to read only property '__esModule' of object '[object Object]'

### ISSUE THREAD

**Issue by BaumVitaly** [ReactTraining/react-router#6649]
### Actual Behavior: Test failed

This is from exported component:

![Screenshot from 2019-03-20 01-31-01](https://user-images.githubusercontent.com/30584905/54632112-f57f3e80-4aaf-11e9-9647-8d8fc740e203.png)

And test failed in import stage in test file

**Comment by timdorr** [comment]
This indicates you have an older version of react-router in your node_modules. Make sure it's updated as well.

**Comment by PrimarchAlpharius** [comment]
I'm having this issue as well and both react-router and react-router-dom are version 5.0.0

**Comment by StringEpsilon** [comment]
@PrimarchAlpharius run `npm ls react-router`. You should only get one entry back.

If you get multiple entries, that is the cause of your issue.

**Comment by PrimarchAlpharius** [comment]
@StringEpsilon Thanks for the quick reply. I've run it and it's returning - (empty), though react-router is definitely installed.
Running` npm ls react-router-dom` does return one entry.
The router is working fine in the actual app, it's just jest that's failing, and only some tests at that.

**Comment by StringEpsilon** [comment]
You might want to ask [StackOverflow ](https://stackoverflow.com/questions/tagged/react-router)or on the [`react-router` channel on Reactiflux. ](https://www.reactiflux.com/).

I have the suspicion that this is a babel-jest / jest issue in your setup. So you should prepare a reproduction repository or a sandbox.

**Comment by cif** [comment]
TL;DR, [Jest does not yet support ES6 modules](https://github.com/facebook/jest/issues/4842) so the CJS modules are being used instead.

This problem stems from https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/index.js#L1

Which transpiles to: 
`Object.keys(reactRouter).forEach(function (key) { exports[key] = reactRouter[key]; });`
in `react-router-dom/cjs/react-router-dom.js`

Because CJS modules prefix this code to them for interop:
`Object.defineProperty(exports, '__esModule', { value: true });`

`TypeError: Cannot assign to read only property '__esModule' of object '[object Object]'` 
is thrown because the require of `react-router` CJS also contains that prefix, meaning the __esModule key defined on `react-router` is read-only. 

Far as solving this, it would probably make the most sense to explicitly state which props of react-router react-router-dom wants to export as well, or remove the problematic `export *` and require users to import what they need from react-router, though it is convenient.  

This looks like a worse idea, but It looks like rollup now has the ability to disable the CJS / es module interop: https://github.com/rollup/rollup/issues/708
